docs/developer-guide/working-with-plugins.md

```diff
@@ -246,4 +246,33 @@ exports.parseForESLint = function(code, options) {
 
 ```
 
+### The AST specification
 
+The AST that custom parsers should create is based on [ESTree](https://github.com/estree/estree) basically. The AST requires some additional properties about detail information of the source code.
+
+#### All nodes:
+
+All nodes must have `range` property.
+
+* `range` (`number[]`) is the array of two numbers. Both numbers are a 0-based index which is the position in the array of source code characters. The first is the start position of the node, the second is the end position of the node. `code.slice(node.range[0], node.range[1])` must be the text of the node. This range does not include spaces which are around the node.
```

Comment:
The first sentence should probably end with "is an array of two numbers" (replace the with an).

Do we need to make a note about parentheses surrounding the node, in some cases? I'm fairly sure parentheses are usually excluded also from the range, unless they're part of the grammar (e.g., arrow function parameters).